src/enclave/main.cpp

```diff
@@ -249,15 +249,25 @@ extern "C"
       return CreateNodeStatus::EnclaveInitFailed;
     }
 
-    auto status = enclave->create_new_node(
-      start_type,
-      std::move(cc),
-      node_cert,
-      node_cert_size,
-      node_cert_len,
-      service_cert,
-      service_cert_size,
-      service_cert_len);
+    CreateNodeStatus status = EnclaveInitFailed;
+
+    try
+    {
+      status = enclave->create_new_node(
+        start_type,
+        std::move(cc),
+        node_cert,
+        node_cert_size,
+        node_cert_len,
+        service_cert,
+        service_cert_size,
+        service_cert_len);
+    }
+    catch (...)
+    {
+      delete enclave;
```

Comment:
It looks like we never delete this object in the normal operation path, after we've stored it in `static std::atomic<enclave::Enclave*> e;`. Should we also be deleting it on line 304?

Reply:
I tried that at some point but it didn't make a difference in the memory leaks I saw. I've mostly been looking at unsuccessful enclave creation though. I will add it anyways, just to be on the safe side.